syntax = "proto3";

import 'files/common/common.proto';

message HeroSkills {
  repeated HeroSkill hero_skills = 1;
  message HeroSkill {
    ObjectIdentity identity = 1;
    HeroSkillInfo info = 2;
  }

  message HeroSkillInfo {
    string skill_type = 5;
    string image = 7;
    repeated HeroSkillValue values = 8;
    int32 unlock_level = 9;
  }

  message HeroSkillValue {
    ObjectIdentity stat = 1;
    ObjectIdentity progression = 3;
    double star_skill_value = 4;
  }
}
